admin/class-manage-fonts.php

```diff
@@ -296,7 +305,82 @@ function ( $font_family ) use ( $font_slug ) {
 			get_stylesheet_directory() . '/theme.json',
 			$theme_json_string
 		);
+	}
+
+	function manage_font_license( $font_name, $file_name ) {
+		if ( ! $font_name ) {
+			return;
+		}
+
+		// Get theme readme.txt
+		$readme_file          = get_stylesheet_directory() . '/readme.txt';
+		$readme_file_contents = file_get_contents( $readme_file );
+
+		if ( ! $readme_file ) {
+			return;
+		}
+
+		// Format font name for font source link
+		$google_formatted_font_name = str_replace( ' ', '+', $font_name );
+
+		// If file_name exists, then add font license to readme.txt
+		if ( 'remove' !== $file_name && is_string( $file_name ) ) {
+			// Check that the font is not already credited in readme.txt
+			if ( false === stripos( $readme_file_contents, $font_name ) ) {
+				// Require php-font-lib
+				require_once( __DIR__ . '/../includes/FontLib/Autoloader.php' );
+
+				// Build font source URL
+				$font_source = 'https://www.google.com/fonts/specimen/' . $google_formatted_font_name;
 
+				// Get license info from font file
+				$font = \FontLib\Font::load( get_stylesheet_directory() . '/assets/fonts/' . $file_name );
+				$font->parse();
+				$license_info = $font->getNameTableString( 13 ) ? $font->getNameTableString( 13 ) : '';
+
+				if ( stripos( $font->getNameTableString( 13 ), 'SIL Open Font License' ) ) {
```

Comment:
Could get all the information from the file metadata instead of searching for specific things?

Reply:
I mean that you can extract the entire license information from the metadata tables of the font file asset. 
Example:
![image](https://user-images.githubusercontent.com/1310626/232865567-5ead5fdf-716d-47ea-8e11-983c8a912311.png)

You can use this online font metadata inspector to look for the name of the metadata property: https://opentype.js.org/font-inspector.html

Reply:
Ah yes, that's what I'm using the FontLib library for here, but the information for the SIL license is a lot longer and looks different from the usual descriptions we use for the fonts in WP themes. So I decided to add a check for the SIL license and then replace it with information in our usual format.